libvast/src/format/csv.cpp

```diff
@@ -416,7 +422,7 @@ caf::error reader::read_impl(size_t max_events, size_t max_slice_size,
       return false;
     } else {
       return lines_->next_timeout(
-        vast::defaults::import::shared::partial_slice_read_timeout);
+        std::chrono::duration_cast<std::chrono::milliseconds>(read_timeout_));
```

Comment:
Using `read_timeout_` here means that we can wait up to **double** the time specified as read_timeout until we actually get the data, which at 10s is a noticable difference.

Also, we should keep in mind that using such a large timeout will increase input lag and it willl increase the likelihood of seeing "double lines" when importing over a socket, that is the last line of some connection w/o newline concatenated with the first line of the next connection. I'm not sure if this is better or worse in the real world than the current behaviour, but we should monitor it and be prepared to separate the two timeouts if it proves to be a problem (i.e. use a short timeout for the line reading and a large one for the `start + read_timeout` check below)

Reply:
Where do you see the doubling up?

Reply:
I totally overlooked this :eyes:.

Reply:
We talked about this via Slack and it's really a matter of documentation. I'll add a note saying that it can be doubled up in the very worst case.

Reply:
The read timeout is now dynamically updated: `read_timeout_ - (now - start)`. This appears to work well in practice.